fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -214,6 +283,42 @@ public Void visitArrowExpr(ArrowExpr node, Scope scope) {
             return null;
         }
 
+        @Override
+        public Void visitLambdaFunction(LambdaFunction node, Scope scope) {
+            ExpressionAnalyzer.analyzeExpression(node.getChild(0), this.analyzeState, scope, this.session);
+            // construct a new scope to analyze the lambda function
+            Scope lambdaScope = scope.getLambdaScope();
+            ExpressionAnalyzer.analyzeExpression(node.getChild(1), this.analyzeState, lambdaScope, this.session);
+            node.setType(Type.FUNCTION);
+            scope.clearLambdaArguments();
+            return null;
+        }
+
+        @Override
+        public Void visitLambdaArguments(LambdaArguments node, Scope scope) {
+            if (scope.getLambdaInputs().size() == 0) {
+                throw new SemanticException("Lambda Functions can only be used in high-order functions.");
+            }
+            List<String> names = node.getNames();
+            Set<String> set = new HashSet<>();
+            for (String name : names) {
+                if (set.contains(name)) {
+                    throw new SemanticException("Lambda argument: " + name + " is duplicated.");
+                }
+                set.add(name);
+            }
+            if (scope.getLambdaInputs().size() != names.size()) {
+                throw new SemanticException("Lambda arguments should equal to lambda input arrays.");
+            }
+            // bind argument names with lambda inputs
+            for (int i = 0; i < names.size(); ++i) {
```

Comment:
Why LambdaArguments need names?

Reply:
help to construct the lambda scope's fields, then lambda expressions can recognize lambda arguments by the name.